rosplan_demos_interfaces/rosplan_interface_mapping/src/RPSimpleMapServer.cpp

```diff
@@ -1,126 +1,191 @@
-		std::string line;
-		int curr,next;
-		while(std::getline(infile, line)) {
-			// read waypoint
-			curr=line.find("[");
-			std::string name = line.substr(0,curr);
-
-			// instance
-			rosplan_knowledge_msgs::KnowledgeUpdateService updateSrv;
-			updateSrv.request.update_type = rosplan_knowledge_msgs::KnowledgeUpdateService::Request::ADD_KNOWLEDGE;
-			updateSrv.request.knowledge.knowledge_type = rosplan_knowledge_msgs::KnowledgeItem::INSTANCE;
-			updateSrv.request.knowledge.instance_type = "waypoint";
-			updateSrv.request.knowledge.instance_name = name;
-			update_knowledge_client.call(updateSrv);
-
-			// data
-			geometry_msgs::PoseStamped pose;
-			pose.header.frame_id = fixed_frame;
-			parsePose(pose, line);
-			std::string id(message_store.insertNamed(name, pose));
-			db_name_map[name] = id;
-
-			// save here for viz
-			Waypoint* wp = new Waypoint(name, pose.pose.position.x, pose.pose.position.y);
-			waypoints[wp->wpID] = wp;
-		}
-		infile.close();
-
-		// publish visualization
-		publishWaypointMarkerArray(nh);
-	}
+    RPSimpleMapServer::RPSimpleMapServer() : nh_("~") {
+
+        // knowledge interface
+        std::string kb;
+        nh_.param<std::string>("knowledge_base", kb, "knowledge_base");
+        std::stringstream ss;
+        ss << "/" << kb << "/update";
+        update_knowledge_client_ = nh_.serviceClient<rosplan_knowledge_msgs::KnowledgeUpdateService>(ss.str());
+        ros::service::waitForService(ss.str(), ros::Duration(20));
+
+        // query params from param server
+        nh_.param<std::string>("fixed_frame", fixed_frame_, "map");
+
+        // visualisation
+        ss.str("");
+        ss << "/" << kb << "/viz/waypoints";
+        waypoints_pub_ = nh_.advertise<visualization_msgs::MarkerArray>(ss.str(), 10, true);
+    }
+
+    /*-----------*/
+    /* build PRM */
+    /*-----------*/
+
+    /**
+    * parses a pose with yaw from strings: "[f, f, f]"
+    */
+    void RPSimpleMapServer::parsePose(geometry_msgs::PoseStamped &pose, std::string line) {
+
+        int curr,next;
+        curr=line.find("[")+1;
+        next=line.find(",",curr);
+
+        pose.pose.position.x = (double)atof(line.substr(curr,next-curr).c_str());
+        curr=next+1; next=line.find(",",curr);
+
+        pose.pose.position.y = (double)atof(line.substr(curr,next-curr).c_str());
+        curr=next+1; next=line.find("]",curr);
+
+        float theta = atof(line.substr(curr,next-curr).c_str());
+        tf::Quaternion q;
+        q.setEuler(theta, 0 ,0);
+
+        pose.pose.orientation.x = q.x();
+        pose.pose.orientation.y = q.z();
+        pose.pose.orientation.w = q.w();
+        pose.pose.orientation.z = q.y();
+    }
+
+    bool RPSimpleMapServer::getWPCoordinates() {
+
+        // function that gets an unknown amount of waypoints from param server under a specific namespace
+
+        XmlRpc::XmlRpcValue waypoint_list;
+
+        if (!nh_.getParam("/waypoints", waypoint_list)) {
```

Comment:
Here the parameter is called something different

Reply:
RPSimpleMapServer.cpp  removed as it is actually unused